bot/cogs/color.py

```diff
@@ -385,6 +387,42 @@ async def set_color(
             )
             return await interaction.followup.send(embed=embed)
 
+    @app_commands.command(name='icon', description='Attach a icon on custom role.')
+    @app_commands.describe(
+        role='The role you want to attach the icon.',
+        icon='The icon you want to attach to the role. Must be a PNG or JPEG(JPG included) file (Maximum 256 KB).',
+    )
+    async def set_icon(
+        self, interaction: Interaction, role: discord.Role, icon: discord.Attachment
+    ) -> None:
+        await interaction.response.defer()
+        if role not in interaction.user.roles:
+            return await interaction.followup.send("❌ Please use the role first.", ephemeral=True)
```

Comment:
This is an efficient way to check the role
https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.get_role

Reply:
let me check